src/anyon/qpu_interface.jl

```diff
@@ -336,8 +336,17 @@ abstract type AbstractQPU end
 get_metadata(qpu::AbstractQPU) = 
     throw(NotImplementedError(:get_metadata,qpu))
 
-get_native_gate_types(qpu::AbstractQPU) = 
-    throw(NotImplementedError(:get_native_gate_types,qpu))
+is_native_gate(qpu::AbstractQPU,::AbstractGate) = 
+    throw(NotImplementedError(:is_native_gate,qpu))
+
+is_native_circuit(qpu::AbstractQPU,::AbstractGate) =
+    throw(NotImplementedError(:is_native_circuit,qpu))
+
+print_connectivity(qpu::AbstractQPU,io::IO=stdout) = 
```

Comment:
This cannot be implemented for all QPUs. What does the connectivity of a VirtualQPU look like?